flyteplugins/go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -28,12 +29,14 @@ import (
 )
 
 type GetClientFunc func(ctx context.Context, agent *Agent, connectionCache map[*Agent]*grpc.ClientConn) (service.AsyncAgentServiceClient, error)
+type GetAgentMetadataClientFunc func(ctx context.Context, agent *Agent, connCache map[*Agent]*grpc.ClientConn) (service.AgentMetadataServiceClient, error)
 
 type Plugin struct {
 	metricScope     promutils.Scope
 	cfg             *Config
 	getClient       GetClientFunc
 	connectionCache map[*Agent]*grpc.ClientConn
+	agentRegistry   map[string]map[bool]*Agent // map[taskType][isSync] => Agent
```

Comment:
Where is this used? Or would that be addressed in follow-up PRs?

Reply:
Yes, this will have follow up PRs, we want to make one agent can execute both sync and async task.
Please leave more questions if this is unclear.
Thank you!

Reply:
update: we will remove the  second dimension, it will improve user experience more in the future, and we will have follow-up PRs for this Registry, thank you.